src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java

```diff
@@ -275,6 +282,69 @@ private void updateSubfieldsMapWithItemLoanTypeSubfield(Map<String, Object> subF
     }
   }
 
+   /**
+   * Constructs field with subfields which is build from holdings record data. Constructed field has tag number = 999 and both
+   * indicators has ' ' value.
+   *
+   * @param jsonData                    - json of single item or holding
+   * @param marcRecordFields            - fields list to be updated with new one
+   * @param suppressedRecordsProcessing - include suppressed flag in 999 field?
+   */
+  private void updateFieldsWithHoldingsRecordField(JsonObject jsonData,
+                                                   List<Object> marcRecordFields,
+                                                   boolean suppressedRecordsProcessing) {
+    Map<String, Object> holdingsRecordSubFields = constructHoldingsRecordSubFieldsMap(jsonData);
+    int subFieldValue = BooleanUtils.isFalse(jsonData.getBoolean(INVENTORY_SUPPRESS_DISCOVERY_FIELD)) ? 0 : 1;
```

Comment:
Should we have another `calculateDiscoverySuppressedSubfieldValue` here?

```
/**
   * Calculates discovery suppressed subfield value.
   *
   * @param itemData - json of single item
   * @return subfield value
   */
  private int calculateDiscoverySuppressedSubfieldValue(JsonObject itemData) {
    if (isNull(itemData.getBoolean(INVENTORY_SUPPRESS_DISCOVERY_FIELD))) {
      return 0;
    }
    return BooleanUtils.isFalse(itemData.getBoolean(INVENTORY_SUPPRESS_DISCOVERY_FIELD)) ? 0 : 1;
  }
  ```

Reply:
It does not appear that they are doing this [upstream](https://github.com/folio-org/mod-oai-pmh/blob/v3.10.0/src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java#L219), so I vote "no".

Reply:
It is an additional null check and that is just a place missed to reduce redundancy. I suggest we do not do the same and use the method created to perform the same logic in one place with the additional null check.